twitter_scrooge/twitter_scrooge.bzl

```diff
@@ -312,7 +312,7 @@ def _scrooge_aspect_impl(target, ctx):
 
         src_jars = depset([scrooge_file])
         output = _compiled_jar_file(ctx.actions, scrooge_file)
-        outs = depset([output])
+        outs = depset([output], transitive = [d[ScroogeAspectInfo].output_files for d in ctx.rule.attr.deps])
```

Comment:
I don't think this is the right way. Often, these providers explicitly propagate a transitive output as well. I think we should add that and include it there. Then when we create the scala struct, we would add it in there.

That said, the scala struct thing is a hack, and we are supposed to only be able to use JavaInfo eventually. I don't know what the current state of the art is there though since bazel was constantly churning on JavaInfo and intellij plugin was behind on it.